http_check/check.py

```diff
@@ -224,17 +223,7 @@ def send_status_down(loginfo, message):
                 self.warning("Skipping SSL certificate validation for %s based on configuration"
                              % addr)
 
-            instance_proxy = self.proxies.copy()
-
-            # disable proxy if necessary
-            if skip_proxy:
-                instance_proxy.pop('http')
-                instance_proxy.pop('https')
-            elif self.proxies['no']:
-                for url in self.proxies['no'].replace(';', ',').split(","):
-                    if url in parsed_uri.netloc:
-                        instance_proxy.pop('http')
-                        instance_proxy.pop('https')
+            instance_proxy = config_proxy_skip(self.proxies.copy(), addr, skip_proxy=skip_proxy)
```

Comment:
Since we're passing copies of `self.proxies` to `requests` and not really mutating it, we should be fine with that approach of moving that logic to a method in AgentCheck, we'll have to call `get_no_proxy_env` on every iteration for each instance (probably the reason behind the current logic), but it's a cheap call, so there's not much harm done.

I've also followed your advice and cleaned up the interface a little bit.  In the current approach, we pass the instance and the address, and we get `skip_url` from the instance dict. We could still argue if passing instance is necessary (but since the method is called get_instance_proxy, it feels like a natural parameter).

Updated the dd-agent side as well.